### ISSUE TITLE
Support multiple HTTP parameter with same key

### ISSUE THREAD

**Issue by klaus0x7c4** [DEGoodmanWilson/luna#60]
It is common you cases like: `http://example.com/action?id=a&id=b`
The current implementation uses a map for the parameter list. So you currently only get the value `id=b` from the example above.

**Comment by DEGoodmanWilson** [comment]
That is true. [RFC 3986](https://tools.ietf.org/html/rfc3986#section-3.4) doesn’t define behavior for this case, in fairness.

Still, this seems like something we should support. How would you envision the Luna API behaving in cases with multiple Parma’s with the same key?

**Comment by klaus0x7c4** [comment]
The easiest way for luna would be to keep a list or vector of string pairs instead of the map. Or a map like: `std::map<std::string, std::vector<std::string>>`. But all that seems to be a breaking change for luna.

The meaning of several identical keys must be clarified by the user in the request handler (or another callback?).

Or you have access to the unparse parameter list next to the map. But I think that's just a quick and dirty solution.

It's hard for me to say what's the right way. The simple vector of string pairs would be enough for me for now.

**Comment by DEGoodmanWilson** [comment]
What about `std::multimap`? Still a breaking change, but perhaps less drastic?

**Comment by klaus0x7c4** [comment]
I didn't know the multimap yet. That sounds good.
Maybe the `std::unordered_multimap` is also an option. 
If I understood it correctly, it is a choice between performance and memory usage.

**Comment by DEGoodmanWilson** [comment]
Yes, we should I think prefer the performance of the std::unordered_multimap. I think _in general_ the number of parameters will be small enough that memory is not an issue—although we might like to gird our loins against DoS attacks that attempt large numbers of query params in the future.
